services/graphql-server/src/user/user-service.js

```diff
@@ -15,6 +15,7 @@ const UserService = class UserService {
       accountNonLocked: true,
       credentialsNonExpired: true,
       enabled: true,
+      password: { $exists: true },
```

Comment:
There should also be a check that the `hash` is not empty, before executing the `replace` call. The `{ $exists: true }` criteria will still return `null` passwords.